### ISSUE TITLE
ZeroTier 1.12.0 - default route issue

### ISSUE THREAD

**Issue by vint2k** [zerotier/ZeroTierOne#2109]
Hi
We have an issue with full-tunnel zerotier network and default route.

For example, we connected to WiFi network at office and connected to the full-tunnel network:
We close the laptop and go home (during this time the laptop go to sleep/hibernation)

When running the laptop at home we observed the following issue:

1. The laptop connected to WiFi, but the Internet doesn't work
2. Routing table has only default route from zerotier (full-tunnel) - dhcp client got network settings, but not apply default route:
```
netstat -nr -f inet
Routing tables

Internet:
Destination        Gateway            Flags           Netif Expire
default            172.16.0.1         UGScg        feth4312
127                127.0.0.1          UCS               lo0
127.0.0.1          127.0.0.1          UH                lo0
127.0.0.1/32       link#26            UCS          feth4312      !
169.254            link#15            UCS               en0      !
172.16/24          link#26            UC           feth4312      !
172.16.0.1         link#26            UHRLWIir     feth4312     17
172.16.0.255       ff:ff:ff:ff:ff:ff  UHLWbI       feth4312      !
192.168.81         link#15            UCS               en0      !
192.168.81.1/32    link#15            UCS               en0      !

```
3. Disconnect full-tunnel network - default route from zerotier disappears and routing table hasn't any other default route:

```
netstat -nr -f inet
Routing tables

Internet:
Destination        Gateway            Flags           Netif Expire
127                127.0.0.1          UCS               lo0
127.0.0.1          127.0.0.1          UH                lo0
169.254            link#15            UCS               en0      !
192.168.81         link#15            UCS               en0      !
192.168.81.1/32    link#15            UCS               en0      !

```
5. You must to reconnect your network to get network settings again
6. Connect full-tunnel network again if you need

Reproduced on MacBook Pro 14 (MacOS 13.4.1/13.5.1) and ZeroTier v1.12.0

Please fix ASAP.

**Comment by ialshaev** [comment]
We have exactly the same issue happened after client version upgrade.
Pretty much annoying.

+1

**Comment by laduke** [comment]
Thanks for writing. Default Route worked reliably for you on the last version, on macos?

**Comment by vint2k** [comment]
Hi @laduke 
Only ipv4 for tunnel and physical networks.
We wait about ~10-15 minutes after connect to the WiFi and nothing changed, only zt daemon just go OFFLINE status after ~5min.

Maybe fix [#2052](https://github.com/zerotier/ZeroTierOne/pull/2052) not work correctly?

**Comment by laduke** [comment]
Possibly related. ZeroTier _shouldn't_ be deleting your en0 default route. I'm curious why it's not coming back on it's own.

**Comment by vint2k** [comment]
I have sent some dumps via service desk.

**Comment by vint2k** [comment]
@laduke hi
any updates?

**Comment by laduke** [comment]
Just working on testing and releasing #2116 and #2110 

If possible let us know if 2116 doesn't get confused about routes when you switch physical networks.

**Comment by vint2k** [comment]
#2116 works well

**Comment by laduke** [comment]
thanks!

**Comment by vint2k** [comment]
I found new issue 
The details have been sent via service desk.

**Comment by laduke** [comment]
try again :)

**Comment by vint2k** [comment]
@laduke everything is working well!
thanks!

**Comment by laduke** [comment]
fixed in 1.12.2. Thanks!

**Comment by codeachange** [comment]
the problem comes back on MacOs Tahoe !
